compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java

```diff
@@ -302,6 +305,17 @@ private void writeFunction(ByteBuf buf, BIRTypeWriter typeWriter, BIRInstruction
         buf.writeBytes(birbuf.nioBuffer().array(), 0, length);
     }
 
+    private void writeFunctionsGlobalVarDependency(ByteBuf buf, BIRNode.BIRFunction birFunction) {
+        List<Integer> globalVarBuf = new LinkedList<>();
+
+        for (BIRNode.BIRVariableDcl var : birFunction.dependentGlobalVars) {
+            globalVarBuf.add(addStringCPEntry(var.name.value));
+        }
+
+        buf.writeInt(globalVarBuf.size());
+        globalVarBuf.forEach(buf::writeInt);
```

Comment:
this can be simplified now right? you don't need an intermediate list, but we can start writing to the buf directly from the loop?